data/calculated_parser/functions.py

```diff
@@ -224,7 +224,9 @@ def sspeed(depth: Union[np.ndarray, xr.Variable],
 
     if salinity.shape != press.shape:
         # pad array shape to match otherwise seawater freaks out
-        press = press[..., np.newaxis]
+        for ax, val in enumerate(salinity.shape):  
+            if ax > press.ndim - 1 or press.shape[ax] != val:
+                press = np.expand_dims(press, axis=ax)
 
     speed = seawater.svel(salinity, temperature, press)
```

Comment:
Not 100% happy with how I did this but it works. The old method added an axis to the end of the array but when we're requesting data over a time range we may also need to pad the beginning so that `press` can be broadcast against `temperature` and `salinity`. For example, if sub-setting GIOPS sound speed salinity shape may be `(3, 50, 3, 12)` and press shape `(50, 3)`. Here we'd us expand dims to give press shape `(1, 50, 3, 1)` which is broadcastable.

Reply:
yeah I see what you're doing here and I can't think of a cleaner way of doing it. The benefit of your method is the manipulation of empty axes which is very lean on memory (as opposed to filling the empty space with 0's).

Frankly I'd put your giops example scenario into the code comment. It instantly enabled me to understand what you're solving for here. Not everyone's first instinct is to pull up the git blame lol

Reply:
Fair! haha. Just added a comment to explain this a bit.